src/lib/spinel/spinel.h

```diff
@@ -4744,6 +4744,16 @@ enum
      */
     SPINEL_PROP_INFRA_IF_STATE = SPINEL_PROP_INFRA_IF__BEGIN + 1,
 
+    /// Received ICMP6 ND message on the infrastructure interface.
+    /** Format: `L6d`
+     * Type: Write-only
+     *
+     * `L`: The infrastructure interface index.
+     * `6`: The IP6 source address of the received message.
+     * `d`: The data of the received message.
+     */
+    SPINEL_PROP_INFRA_IF_RECV_ICMP6_ND = SPINEL_PROP_INFRA_IF__BEGIN + 2,
```

Comment:
should we make this general ICMPv6 packets?
```suggestion
    SPINEL_PROP_INFRA_IF_RECV_ICMP6 = SPINEL_PROP_INFRA_IF__BEGIN + 2,
```

Reply:
I don't think so. Because after NCP receives this message, `otPlatInfraIfRecvIcmp6Nd` should be called which only handles ND messages.

If we want to support other Icmp6 messages in the future,  we will need to add extra parsing and handling code in NCP (check if this is a ND or other ICMP6 messages), which introduces extra dependency and adds the complexity of NCP. Because OT core can already handle these, we should let OT core handle them. Thoughts?

Reply:
@bukepo As discussed, use this property for all ICMP6 messages instead of only for ND messages. Added a field in this property for the ICMP6 message type.

Reply:
Sounds okay to me but I am just curios what are some use-cases of non-ND type ICMPv6 messages from infra-if that may be useful?

Reply:
For now there is no use case of non-ND type ICMPv6 messages from infra-if. I think we can probably leave the extensibility here for the future.

Reply:
@abtink Currently, a dedicated host socket is created for joining any multicast group that a node in Thread network subscribes to(see [code](https://github.com/openthread/openthread/blob/a9805ed93a5c6518ae351bf050ea339f4bbe1052/src/posix/platform/multicast_routing.cpp#L166)). It is a trick to leverage Linux kernel's existing MLDv2 implementation to send and process MLDv2 messages. However, the trick actually causes the host itself to subscribe to those multicast groups while the host itself probably isn't interested in them. As a result, unwanted multicast traffic is delivered to the input chain of the host's IPv6 stack. A more elegant way may be handling the MLDv2 messages by OpenThread itself, which avoids creating that host socket. That would require OT to receive and send MLDv2 messages which are also based on ICMPv6.